spring-rest2ts-jackson-js/src/main/java/com/blueveery/springrest2ts/converters/JacksonAnnotationsConversionToJacksonJs.java

```diff
@@ -0,0 +1,208 @@
+            case NAME:
+                return new TSLiteral("", TypeMapper.tsString, typeIdResolver.apply(javaType, javaRoot));
+            default:
+                throw new IllegalStateException(jsonTypeInfo.use() + " is not supported");
+        }
+    }
+
+    private String getMinimalClassName(Class javaType, Class javaRoot) {
+        if ( javaType.getName().startsWith(javaRoot.getPackage().getName())) {
+            return javaType.getName().replaceFirst(javaRoot.getPackage().getName(), "");
+        }
+        return javaType.getName();
+    }
+
+    private Class findInheritanceRoot(Class javaType) {
+        Class superclass = javaType.getSuperclass();
+        if (superclass != null && superclass != Object.class && TypeMapper.map(superclass) != TypeMapper.tsAny) {
+            if (superclass.getAnnotation(JsonTypeInfo.class) == null) {
+                return findInheritanceRoot(superclass);
+            } else {
+                return superclass;
+            }
+        }
+        return null;
+    }
+
+    private TSDecorator addJsonSubTypesAnnotation(TSClass tsClass) {
+        TSDecorator jsonSubTypesDecorator = new TSDecorator(jsonSubTypesFunction);
+        TSJsonLiteral literal = new TSJsonLiteral();
+        literal.getFieldMap().put("types", new TSLiteralArray());
+        jsonSubTypesDecorator.getTsLiteralList().add(literal);
+        tsClass.getTsDecoratorList().add(jsonSubTypesDecorator);
+        tsClass.addScopedTypeUsage(jsonSubTypesFunction);
+        return jsonSubTypesDecorator;
+    }
+
+    private void addJsonTypeInfoAnnotation(TSClass tsClass, JsonTypeInfo jsonTypeInfo) {
+        TSDecorator jsonTypeInfoDecorator = new TSDecorator(jsonTypeInfoFunction);
+        TSJsonLiteral jsonTypeInfoLiteral = new TSJsonLiteral();
+        jsonTypeInfoDecorator.getTsLiteralList().add(jsonTypeInfoLiteral);
+        tsClass.addScopedTypeUsage(jsonTypeInfoFunction);
+
+        convertJsonTypeInfoUse(tsClass, jsonTypeInfo, jsonTypeInfoLiteral);
+        if (convertJsonTypeInfoInclude(tsClass, jsonTypeInfo, jsonTypeInfoLiteral)) {
+            convertJsonTypeInfoProperty(jsonTypeInfo, jsonTypeInfoLiteral);
+        }
+
+        tsClass.getTsDecoratorList().add(0, jsonTypeInfoDecorator);
+    }
+
+    private void convertJsonTypeInfoProperty(JsonTypeInfo jsonTypeInfo, TSJsonLiteral jsonTypeInfoLiteral) {
+        String propertyName = jsonTypeInfo.property();
+        if("".equals(propertyName)) {
+            propertyName = jsonTypeInfo.use().getDefaultPropertyName();
+        }
+        if (!"@type".equals(propertyName)) {
+            jsonTypeInfoLiteral.getFieldMap().put("property", new TSLiteral("", TypeMapper.tsString, propertyName));
+        }
+    }
+
+    private void convertJsonTypeInfoUse(TSClass tsClass, JsonTypeInfo jsonTypeInfo, TSJsonLiteral jsonTypeInfoLiteral) {
+        if (Arrays.asList(JsonTypeInfo.Id.NAME, JsonTypeInfo.Id.CLASS, JsonTypeInfo.Id.MINIMAL_CLASS).contains(jsonTypeInfo.use())) {
```

Comment:
For enum values, a faster solution is to use EnumSet and also you should consider extracting this variable as a static final class field instead of creating each time array in function.     
        ```final EnumSet<JsonTypeInfo.Id> jsonInfoTypes = EnumSet.of(JsonTypeInfo.Id.NAME, JsonTypeInfo.Id.CLASS, JsonTypeInfo.Id.MINIMAL_CLASS);```

Reply:
EnumSet => ok, but extracting to field will reduce readability, performance is not so important as genrator is super quick